optuna/terminator/improvement/evaluator.py

```diff
@@ -35,6 +36,7 @@ def evaluate(
         pass
 
 
+@experimental_class("3.2.0")
 class RegretBoundEvaluator(BaseImprovementEvaluator):
```

Comment:
`BaseImprovementEvaluator` is not marked as experimental. Do we keep the compatibility from now?

If we add the experimental decorator to both `RegretBoundEvaluator` and `BaseImprovementEvaluator`, only the warning message of `RegretBoundEvaluator` is shown. So, the duplicated warning messages do not suffer users.

Reply:
Good catch. As long as the message is not duplicated, let's make the base classes experimental as well.

I also changed related class and functions.
- `BaseGaussianProcess` is now an experimental class, since it is exposed to the user and guaranteeing the compatibility is still risky.
- `min_ucb` and `min_lcb` is now private functions, since their exposure is not required.

Reply:
Thank you for your update. Sounds reasonable to me!